docs/data-type-support.rst

```diff
@@ -0,0 +1,65 @@
+.. meta::
+   :description: rocPRIM API library data type support
+   :keywords: rocPRIM, ROCm, API library, API reference, data type, support
+
+.. _data-type-support:
+
+******************************************
+Data type support
+******************************************
+
+* Supported input and output types.
+
+  .. list-table:: Supported Input/Output Types
+    :header-rows: 1
+    :name: supported-input-output-types
+
+    *
+      - Input/Output Types
+      - Library Data Type
+      - AMD Supports
```

Comment:
What is the difference between the "library data type" and "amd supports" column? The information provided there seems somewhat redundant, or does "library data type" mean whether it is supported in rocprim and "AMD supports" whether it is supported in hardware on AMD GPUs?

Also is the ❌ for int16_t correct? There are some tests that use it, so I would guess that it's at least somewhat supported

Reply:
i'll look at it